algoliasearch/src/main/java/com/algolia/search/saas/Query.java

```diff
@@ -981,6 +995,96 @@ public TypoTolerance getTypoTolerance()
         return null;
     }
 
+    private static final String KEY_EXACT_ON_SINGLE_WORD_QUERY = "exactOnSingleWordQuery";
+
+    public @NonNull Query setExactOnSingleWordQuery(ExactOnSingleWordQuery type) {
+        if (type == null) {
+            set(KEY_EXACT_ON_SINGLE_WORD_QUERY, null);
+        } else {
+            switch (type) {
+                case NONE:
+                    set(KEY_EXACT_ON_SINGLE_WORD_QUERY, "none");
+                    break;
+                case WORD:
+                    set(KEY_EXACT_ON_SINGLE_WORD_QUERY, "word");
+                    break;
+                case ATTRIBUTE:
+                    set(KEY_EXACT_ON_SINGLE_WORD_QUERY, "attribute");
+                    break;
+            }
+        }
+        return this;
+    }
+
+    public ExactOnSingleWordQuery getExactOnSingleWordQuery()
+    {
+        String value = get(KEY_EXACT_ON_SINGLE_WORD_QUERY);
+        if (value != null) {
+            switch (value) {
+                case "none":
+                    return ExactOnSingleWordQuery.NONE;
+                case "word":
+                    return ExactOnSingleWordQuery.WORD;
+                case "attribute":
+                    return ExactOnSingleWordQuery.ATTRIBUTE;
+            }
+        }
+        return null;
+    }
+
+    private static final String KEY_ALTERNATIVES_AS_EXACT = "alternativesAsExact";
+
+    public @NonNull Query setAlternativesAsExact(List<AlternativesAsExact> types) {
```

Comment:
Why not `AlternativesAsExact[]`?

1. This would be more consistent with the rest of the `Query` class.
2. This would allow using variable arguments, as in:

```query.setAlternativesAsExact(Query.AlternativesAsExact.IGNORE_PLURALS, Query.AlternativesAsExact.MULTI_WORDS_SYNONYM)```